lib/docker/DockerRun.ts

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright © 2018 Atomist, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+    DefaultGoalNameGenerator,
+    ExecuteGoal,
+    ExecuteGoalResult,
+    FulfillableGoalWithRegistrations,
+    GoalInvocation,
+    ImplementationRegistration,
+    IndependentOfEnvironment,
+} from "@atomist/sdm";
+import {
+    DockerRunner,
+    DockerRunnerOptions,
+} from "./DockerRunner";
+
+export interface DockerRunRegistration extends Partial<ImplementationRegistration> {
+    lowerPort?: number;
+    successPatterns: RegExp[];
+    baseUrl?: string;
+    sourcePort: number;
+}
+
+export class DockerRun extends FulfillableGoalWithRegistrations<DockerRunRegistration> {
+    constructor(uniqueName: string = DefaultGoalNameGenerator.generateName("docker-run")) {
```

Comment:
constructor should follow the pattern in the other typed goals. See https://github.com/atomist/sdm-core/blob/master/src/goal/common/Version.ts#L44-L51 for an example